### PR TITLE
Add evidenceSubmissions to claims index endpoint with optimized queries

### PR DESCRIPTION
**Pull request by dfitchett**
- Include evidenceSubmissions in index response when cst_show_document_upload_status flag enabled
  - Optimize to fetch evidence submissions once for both response and metrics reporting
  - Refactor to reuse add_evidence_submissions_to_claims helper for both index and show endpoints
  - Add proper error handling with endpoint-specific logging

# Pull Request Description

Keep your PR as a Draft until it's ready for Platform review. A PR is ready for Platform review when it has a teammate approval and tests, linting, and settings checks pass CI. See [these tips](https://depo-platform-documentation.scrollhelp.site/developer-docs/vets-api-pr-tips) on how to avoid common delays in getting your PR merged.

## Summary

- **This work is behind a feature toggle (flipper): YES** - `cst_show_document_upload_status` and `cst_update_evidence_submission_on_show`
- **Team**: BMT Team 2 (Bee's Knees) - Benefits Management Tools
- **Component Ownership**: Yes, BMT Team 2 owns the maintenance of the Benefits Claims Controller

### Changes Made

This PR refactors evidence submission handling in the `BenefitsClaimsController` to improve performance and maintainability through optimized database queries and better error handling.

#### Key Improvements:

1. **Performance Optimization - Batch Fetching**:
   - **Index Action**: Now fetches all evidence submissions for all claims 
   - **Show Action**: Streamlined evidence submission fetching and processing flow
   - Reduces N+1 query issues when displaying multiple claims

2. **Code Organization & Maintainability**:
   - Extracted `fetch_evidence_submissions` method with centralized error handling
   - Extracted `add_evidence_submissions_to_claims` method to handle adding evidence submissions to claim data
   - Refactored `update_evidence_submissions_for_claim` to focus solely on polling/updating statuses
   - Updated `report_evidence_submission_metrics` to accept evidence submissions directly, eliminating redundant database queries

3. **Enhanced Error Handling**:
   - Graceful degradation when evidence submission fetching fails
   - Graceful degradation when adding evidence submissions to claims fails
   - Structured error logging with consistent format including: endpoint name, claim IDs, error details, and timestamp
   - Frontend already handles missing `evidenceSubmissions` attribute, so failures don't break the user experience

### Why This Solution?

As part of the CST doc status effort, the #index endpoint needs to have access to evidence submissions to display slim alerts for claims that have failed evidence submissions. 

### Success Criteria

The feature flippers remain unchanged, but this refactoring improves:
- Response time for claims index endpoint (reduced database queries)
- Code maintainability through better separation of concerns
- System resilience through improved error handling

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/125469

## Testing done

- [x] **New code is covered by unit tests**

### Old Behavior
- Evidence submissions were fetched individually for each claim
- Metrics reporting made additional database queries for upload status counts
- Error handling was inconsistent between index and show endpoints
- No tests for error scenarios when evidence submission processing fails

#### Unit Tests Added:
1. **Error Handling Tests** (both `index` and `show` endpoints):
   - Tests for when `fetch_evidence_submissions` fails with database errors
   - Tests for when `add_evidence_submissions_to_claims` fails during processing
   - Verifies graceful degradation - requests succeed even when evidence submission processing fails
   - Verifies structured error logging with correct endpoint name and claim IDs

2. **Feature Flag Tests**:
   - Tests that `evidenceSubmissions` are included when feature flag is enabled
   - Tests that `evidenceSubmissions` are NOT included when feature flag is disabled
   - Covers both flipper on and flipper off scenarios as required

3. **Data Integrity Tests**:
   - Verifies evidence submissions are correctly associated with their respective claims
   - Tests filtering of duplicate evidence submissions
   - Tests handling of claims with no evidence submissions (empty array)
   - Tests multiple claims with different evidence submissions

4. **Edge Cases**:
   - Empty evidence submissions array when no records exist
   - Multiple claims each with their own evidence submissions
   - Claims with mixed upload statuses (success, failed, pending)

#### Staging Testing Steps:
1. Start vets-api locally with the feature flags enabled
2. Access the claims index endpoint: `GET /v0/benefits_claims`
   - Verify response includes `evidenceSubmissions` array for each claim
   - Check DataDog logs for evidence submission metrics
3. Access the claims show endpoint: `GET /v0/benefits_claims/:id`
   - Verify response includes `evidenceSubmissions` array
   - Verify metrics are logged correctly
4. Test with feature flag disabled:
   - Toggle `cst_show_document_upload_status` off
   - Verify `evidenceSubmissions` is NOT included in response
5. Verify database query reduction:
   - Check Rails logs for number of queries executed
   - Should see batched queries instead of N+1 pattern

#### Testing Plan for Feature Rollout:
- These changes maintain backward compatibility with existing feature flags
- No additional rollout testing required as feature flags remain unchanged
- Performance improvements are transparent to end users

## Screenshots
_N/A - Backend refactoring with no UI changes_

## What areas of the site does it impact?

**Primary Impact**:
- `GET /v0/benefits_claims` (Index) - Claims listing page in Claim Status Tool
- `GET /v0/benefits_claims/:id` (Show) - Individual claim detail page

**Related Systems**:
- EvidenceSubmission model and database queries
- DataDog metrics for document upload status tracking
- Benefits Documents API integration (unchanged, but query patterns improved)

## Acceptance criteria

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No error nor warning in the console.
- [x] Events are being sent to the appropriate logging solution
- [ ] Documentation has been updated (link to documentation) - _No documentation updates needed for internal refactoring_
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Feature/bug has a monitor built into Datadog (if applicable) - _Existing DataDog metrics maintained_
- [x] If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ] I added a screenshot of the developed feature - _N/A for backend refactoring_

### Areas for Review:

1. **Performance Validation**: Please review the query optimization approach - specifically the batching of evidence submissions in the `index` action. Does the approach scale well with large numbers of claims?

2. **Error Handling Strategy**: The graceful degradation approach returns empty evidence submissions on errors. Is this the desired behavior, or should we consider alternative error handling strategies?

3. **Method Extraction**: Please review the newly extracted methods (`fetch_evidence_submissions`, `add_evidence_submissions_to_claims`) for clarity and reusability.

4. **Test Coverage**: Please verify the test coverage adequately covers edge cases, especially around the error handling scenarios and feature flag combinations.

### Notes:
- This refactoring maintains 100% backward compatibility
- No changes to API response structure when feature flags are not enabled

### CODE REVIEW

File: app/controllers/v0/benefits_claims_controller.rb [hunk 1]
```diff
@@ -305,15 +309,9 @@ def suppress_evidence_requests(claim)
       claim
     end
 
-    def report_evidence_submission_metrics(endpoint, claim_ids)
-      claim_ids = Array(claim_ids) # Ensure it's always an array
+    def report_evidence_submission_metrics(endpoint, evidence_submissions)
+      status_counts = evidence_submissions.group_by(&:upload_status).transform_values(&:count)
```
Comment (RachalCassity) [inline app/controllers/v0/benefits_claims_controller.rb]: Vets-API has a large database, are you sure you want to remove this query from the ActiveRecord level?
